specutils/io/read_fits.py

```diff
@@ -432,9 +438,16 @@ def read_fits_wcs_linear1d(wcs_info, dispersion_unit=None, spectral_axis=0):
     if None in [dispersion_start, dispersion_delta, pixel_offset]:
         raise FITSWCSSpectrum1DError
     dispersion_start += -pixel_offset * dispersion_delta
-    return specwcs.Spectrum1DPolynomialWCS(degree=1, unit=dispersion_unit,
-                                           c0=dispersion_start,
-                                           c1=dispersion_delta)
+    # Log/Linear?
+    dc_flag = wcs_info.affine_transform_dict['DC-FLAG'] 
+    if dc_flag == 1: # Log
```

Comment:
seeing what you are doing here - I would just give `dc_flag` True or False. so you can write `if dc_flag`